third_party/unix/linux/types.go

```diff
@@ -0,0 +1,67 @@
+//go:build ignore
+// +build ignore
+
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+/*
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+Modifies:
+- all the types and consts defined below are addition to the original code
+- which will generate codes in ztypes_linux.go by GOOS=linux ./mkall.sh
+*/
+/*
+Input to cgo -godefs.  See README.md
+*/
+// +godefs map struct_in_addr [4]byte /* in_addr */
+// +godefs map struct_in6_addr [16]byte /* in6_addr */
+package unix
+
+/*
+include <linux/mroute.h>
+
+// copied from /uapi/linux/mroute.h
+// The original struct has union of vifc_lcl_addr and vifc_lcl_ifindex.
+// We only want to vifc_lcl_ifindex here
+struct vifctl_with_ifindex {
+	vifi_t	vifc_vifi;
+	unsigned char vifc_flags;
+	unsigned char vifc_threshold;
+	unsigned int vifc_rate_limit;
+	int            vifc_lcl_ifindex;
+	struct in_addr vifc_rmt_addr;
+};
+*/
+
+import "C"
```

Comment:
@antoninbas I recall you were concerned importing CGO because of something related to arm compilation, would this change concern you?

Reply:
Yes I had general concerns for cross-building and binary distribution. However, I notice that there is a `//go:build ignore` in this file, so isn't this file ignored altogether when building any Antrea package?

Reply:
So after looking at this a bit more, I would say this file is used to generate other "pure Go" files in this directory (see `third_party/unix/ztypes_linux.go`), which are the source files used to build Antrea. @ceclinux can correct me but this file should indeed be ignored and this PR doesn't introduce a cgo "dependency" for building Antrea.

Reply:
Your description is correct. `third_party/unix/linux/types.go` are not get built and used directly in our repo. It is used to generate os-arch-spec "pure Go" files `zsyscall...go`, `zsysnum...go`, `ztypes...go` and `zerrors_linux` by mkall script https://github.com/golang/sys/blob/master/unix/mkall.sh, which will get built eventually.

Reply:
Thanks for the explanation.

Reply:
We should  add some comments to explain this.